src/pair_potential_types.F

```diff
@@ -1795,6 +1830,85 @@ SUBROUTINE pair_potential_nequip_release(nequip)
       END IF
    END SUBROUTINE pair_potential_nequip_release
 
+! **************************************************************************************************
+!> \brief Creates the ALLEGRO potential type
+!> \param allegro ...
+!> \author Teodoro Laino [teo] 11.2005
+! **************************************************************************************************
+   SUBROUTINE pair_potential_allegro_create(allegro)
+      TYPE(allegro_pot_type), POINTER                    :: allegro
+
+      CPASSERT(.NOT. ASSOCIATED(allegro))
+      ALLOCATE (allegro)
+      allegro%allegro_file_name = ""
+      allegro%unit_coords = ""
+      allegro%unit_forces = ""
+      allegro%unit_energy = ""
+      allegro%unit_cell = ""
+   END SUBROUTINE pair_potential_allegro_create
+
+! **************************************************************************************************
+!> \brief Copy two ALLEGRO potential type
+!> \param allegro_source ...
+!> \param allegro_dest ...
+!> \author Teodoro Laino [teo] 11.2005
+! **************************************************************************************************
+   SUBROUTINE pair_potential_allegro_copy(allegro_source, allegro_dest)
+      TYPE(allegro_pot_type), POINTER                    :: allegro_source, allegro_dest
+
+      IF (.NOT. ASSOCIATED(allegro_source)) RETURN
+      IF (ASSOCIATED(allegro_dest)) CALL pair_potential_allegro_release(allegro_dest)
+      CALL pair_potential_allegro_create(allegro_dest)
+      allegro_dest%allegro_file_name = allegro_source%allegro_file_name
+      allegro_dest%allegro_version = allegro_source%allegro_version
+      allegro_dest%rcutsq = allegro_source%rcutsq
+      allegro_dest%unit_coords = allegro_source%unit_coords
+      allegro_dest%unit_coords_val = allegro_source%unit_coords_val
+      allegro_dest%unit_forces = allegro_source%unit_forces
+      allegro_dest%unit_forces_val = allegro_source%unit_forces_val
+      allegro_dest%unit_energy = allegro_source%unit_energy
+      allegro_dest%unit_energy_val = allegro_source%unit_energy_val
+      allegro_dest%unit_cell = allegro_source%unit_cell
+      allegro_dest%unit_cell_val = allegro_source%unit_cell_val
```

Comment:
It could be simpler to just write `allegro_dest = allegro_source`

Reply:
Done, thanks.